swift/Tests/Blockchains/OasisTests.swift

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import XCTest
+import WalletCore
+
+class OasisTests: XCTestCase {
+
+    func testAddress() {
+
+        let key = PrivateKey(data: Data(hexString: "4f8b5676990b00e23d9904a92deb8d8f428ff289c8939926358f1d20537c21a0")!)!
+        let pubkey = key.getPublicKeyEd25519()
+        let address = AnyAddress(publicKey: pubkey, coin: .oasis)
+        let addressFromString = AnyAddress(string: "oasis1qzawzy5kaa2xgphenf3r0f5enpr3mx5dps559yxm", coin: .oasis)!
+
+        XCTAssertEqual(pubkey.data.hexString, "93d8f8a455f50527976a8aa87ebde38d5606efa86cb985d3fb466aff37000e3b")
+        XCTAssertEqual(address.description, addressFromString.description)
+    }
+
+    func testSign() {
+        let privateKey = PrivateKey(data: Data(hexString: "0x4f8b5676990b00e23d9904a92deb8d8f428ff289c8939926358f1d20537c21a0")!)!
+        let input = OasisSigningInput.with {
+            $0.privateKey = privateKey.data
+            $0.transfer = OasisTransferMessage.with {
+                $0.to = "oasis1qrrnesqpgc6rfy2m50eew5d7klqfqk69avhv4ak5"
+                $0.method = "staking.Transfer"
+                $0.gasPrice = 0
+                $0.gasAmount = Data("0".utf8)
+                $0.amount = Data("10000000".utf8)
+                $0.nonce = 0
+                $0.context = "oasis-core/consensus: tx for chain bc1c715319132305795fa86bd32e93291aaacbfb5b5955f3ba78bdba413af9e1"
```

Comment:
what's `context` used for? arbitrary string?

Reply:
It is used to understand the tx data in specific ways depending on the value of that field. For example, 

```
const char context_prefix_tx[] = "oasis-core/consensus: tx for chain ";
const char context_prefix_entity[] = "oasis-core/registry: register entity";
const char context_prefix_node[] = "oasis-core/registry: register node";
const char context_prefix_consensus[] = "oasis-core/tendermint";
const char context_prefix_entity_metadata[] = "oasis-metadata-registry: entity";
```

The possible values are standard.

Reply:
in this case, can we add them as constants?

Reply:
Are these values required?

Reply:
What values? The ones I wrote in my reply? These values are examples of context. I will add them as constants.

Reply:
The strings I wrote in my reply are examples of context values. They are not necessary in the code.

Reply:
empty context tx is also valid?

Reply:
The context needs to be prepended when calculating the digest that will be signed